x509util/extensions.go

```
@@ -500,3 +533,170 @@ func (s *SerialNumber) UnmarshalJSON(data []byte) error {
+							return nil, fmt.Errorf("invalid int value for int-typed SAN OtherName %s", san.Type)
+						}
+						rawBytes, err = asn1.Marshal(i)
+					case "oid":
+						var oidVal asn1.ObjectIdentifier
+						oidVal, err = parseObjectIdentifier(sanValue)
+						if err != nil {
+							return nil, fmt.Errorf("invalid OID value for OID-typed SAN OtherName %s", san.Type)
+						}
+
+						rawBytes, err = asn1.Marshal(oidVal)
+					case "raw":
+						// the raw type accepts a base64 encoded byte array which is passed unaltered into the ASN
+						// marshaller. By using this type users can add ASN1 datatypes manually into templates
+						// to support some unsupported types like BMPString or HEX types
+						rawBytes, err = base64.StdEncoding.DecodeString(sanValue)
+					case "utf8":
+						fallthrough
+					case "ia5":
+						fallthrough
+					case "numeric":
+						fallthrough
+					case "printable":
+						rawBytes, err = asn1.MarshalWithParams(sanValue, valueType)
+					default:
+						// if it's an unknown type, default to printable - but use the entire value specified in case there is a semicolon in the value
+						rawBytes, err = asn1.MarshalWithParams(san.Value, "printable")
+					}
+
+					if err != nil {
+						return nil, fmt.Errorf("could not marshal ASN1 values: %v", err)
+					}
+
+					otherNameValue = asn1.RawValue{FullBytes: rawBytes}
+
+					// OtherName SANs are an ASN1 sequence containing OID and Value
+					otherName := OtherName{
+						OID:   oid,
+						Value: OtherNameValue{V: otherNameValue},
+					}
+					generalNameBytes, _ := asn1.MarshalWithParams(otherName, fmt.Sprintf("tag:%d", nameTypeOtherName))
+					rawValues = append(rawValues, asn1.RawValue{FullBytes: generalNameBytes})
+				} else {
+					return nil, fmt.Errorf("unsupported SAN type %s", san.Type)
+				}
+			}
+
+		}
+
+	}
+
+	// Now marshal the rawValues into the ASN1 sequence, and create an Extension object to hold the extension
+	rawBytes, _ := asn1.Marshal(rawValues)
```

Do not ignore errors, and use errors.Wrap(err, "error marshaling to asn1") to add a stack trace.